### PR TITLE
Feature/cloud 912 okta app

### PR DESCRIPTION
**Pull request by salyateem13**
# Description

Create Okta oauth app, Okta user and admin groups and assignments.

Fixes [#8](https://usxtech.atlassian.net/browse/CLOUD-8)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Tested by creating an example front-end SPA oauth application in Okta. 
```yaml
- Prereqs
    - Clone repo, switch to this branch  `feature/CLOUD-912-okta-app`
    - Create a dummy Okta user to sign in as and test admin group permissions. Or reach out to me for my dummy user password
    - Go to okta preview admin page> Security > API > Tokens and create a token. Save for next step
    -  cd to `./examples/single_page_app_oath`. Open main.tf and change `module.spa_oauth_app.okta_provider_config` with proper values. 
    - run terraform init and apply
- Cases
    - Oath SPA app created.
         - Open Production Okta dashboard and see configurations for RVT web app.
         - Compare configuration of newly created oath app is the same (minus redirect urls) as RVT app. 
    - Admin and User group created and are assigned to newly created app.
         - `app-${var.environment}-${var.name}-administrators` and `app-${var.environment}-${var.name}-users` should exist and are assigned to Okta app named `${var.environment}-${var.name}`
    - Admin group can assign users to 
         - Assign dummy user to admin group. Sign out and sign in as dummy user. Go to groups and try to add users the user group of the newly created Okta app.  
```

- [x] Sanity Testing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added documentation to test

### CODE REVIEW

File: provider.tf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+provider "okta" {
```
Comment (vibindaniel) [inline provider.tf:1]: Can the requirements be a single object?

File: versions.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+terraform {
+    required_version = ">=1.0.0"
+    required_providers {
+        okta = {
+            source = "okta/okta"
+            version = "~> 3.20"
```
Comment (vibindaniel) [inline versions.tf:6]: Change version to `>= 3.20`

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    config = local.configs["${var.type}"]
+}
+
+resource "okta_group" "admin" {
+    name = "app-${var.environment}-${var.name}-administrators"
+}
+
+resource "okta_group" "users" {
+    name = "app-${var.environment}-${var.name}-users"
+
```
Comment (vibindaniel) [inline main.tf:23]: remove empty line

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+resource "okta_group_role" "app_group_admin" {
+    group_id  = okta_group.admin.id
+    role_type = "GROUP_MEMBERSHIP_ADMIN"
+    target_group_list=[okta_group.users.id]
+}
+
+resource "okta_app_oauth" "frontend_spa" {
+    label = "${var.name}-${var.environment}"
```
Comment (vibindaniel) [inline main.tf:33]: "${var.environment}-${var.name}"

**Review by vibindaniel — CHANGES_REQUESTED** [review]
There is no `okta_app_group_assignment` found. Can you add all options from the resource and make some as defaults?

File: main.tf (hunk 4 above)
Comment (salyateem13) [reply main.tf:33]: Was thinking ${var.name}-${var.environment} would be better in UI... see app list sorted by app name rather than by env

**Review by rchengvariant — APPROVED** [review]

**Review by rchengvariant — APPROVED** [review]

**Review by LeeHughsVariant — APPROVED** [review]

**Pull request merged**